packages/bpk-component-datatable/src/BpkDataTable.tsx

```diff
@@ -122,17 +116,17 @@ const BpkDataTable = (props: Props) => {
     );
 
   const onRowClicked = (index: number) => {
-    if (rowSelected === index) {
-      updateRowSelected(undefined);
-    } else {
-      updateRowSelected(index);
-    }
     if (onRowClick !== undefined) {
+      if (rowSelected === index) {
```

Comment:
moved this inside the `if` as previously it was triggered even if `onRowClick` was undefined which meant it still changed the colour to show it as selected